cms/navigation/models.py

```diff
@@ -0,0 +1,115 @@
+
+from django.core.exceptions import ValidationError
+from django.db import models
+from wagtail.admin.panels import FieldPanel
+from wagtail.blocks import CharBlock, ListBlock, PageChooserBlock, StructBlock, URLBlock
+from wagtail.contrib.settings.models import BaseSiteSetting, register_setting
+from wagtail.snippets.models import register_snippet
+
+from cms.core.fields import StreamField
+
+
+# Custom LinkBlock to support both pages and URLs
+class LinkBlock(StructBlock):
+    page = PageChooserBlock(required=False)
+    title = CharBlock(required=False, help_text="Optional. Displayed as the link text.")
+    url = URLBlock(required=False)
+
+    def clean(self, value):
+        value = super().clean(value)
+        if not value.get("page") and not value.get("url"):
+            raise ValidationError("Either a page or a URL must be provided.")
+        return value
+
+    class Meta:
+        icon = "link"
+        label = "Link"
+
+
+# Highlights StructBlock
+class HighlightsBlock(StructBlock):
+    page = PageChooserBlock(required=False)
+    url = URLBlock(required=False)
+    title = CharBlock(
+        required=False, help_text="Optional. Displayed as the link text. Required if adding an external URL."
+    )
+    description = CharBlock(required=True, max_length=50, help_text="E.g., It's never been more important")
+
+    def clean(self, value):
+        value = super().clean(value)
+        if not value.get("page") and not value.get("url"):
+            raise ValidationError("Either a page or a URL must be provided.")
+        return value
+
+    class Meta:
+        icon = "star"
+        label = "Highlight"
+
+
+# Section StructBlock for columns
+class SectionBlock(StructBlock):
+    section_link = LinkBlock(help_text="Main link for this section (Theme pages or external URLs).")
```

Comment:
this may need a separate block definition as the requirement says the PageChooserBlock should be limited to Theme page ([Wagtail docs](https://docs.wagtail.org/en/stable/reference/streamfield/blocks.html#wagtail.blocks.PageChooserBlock))

Reply:
Done!